src/components/ui/ProductDetails.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'evergreen-ui';
+import QuantityBtn from './QuantityBtn';
+import CollapseBtns from './CollapseBtns';
+
+const ProductDetails = ({
+  productvalues,
+}) => {
+  const [productQuntity, setProductQuntity] = useState(1);
+  const onlyPositive = /[1-9]\d?/;
+
+  const validateNewQuantity = (prev) => {
+    if (onlyPositive.test(prev.target.value) || prev.target.value === '') {
+      setProductQuntity(prev.target.value);
+    }
+  };
+
+  const validateSubtraction = () => {
+    if (productQuntity > 1) {
+      setProductQuntity(productQuntity - 1);
+    }
+  };
+
+  const validateSum = () => {
+    if (productQuntity >= 0) {
+      setProductQuntity(productQuntity - 1 + 2);
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+  };
+
+  return (
+    <div className="product-view-container">
+      <div className="product-img-container">
+        <div className="product-img-wrapper">
+          <img data-cy="product-image" alt="product-img" src={productvalues.image} />
+        </div>
+      </div>
+      <div className="product-info-container">
+        <div className="product-info-wrapper">
+          <h2 data-cy="product-info-title" className="product-info-title">{productvalues.title}</h2>
+          <p data-cy="product-info-price" className="product-info-price">
+            {`$${productvalues.price} USD`}
+          </p>
+          <form onSubmit={handleSubmit}>
+            <p>Quantity</p>
+            <QuantityBtn
+              validateNewQuantity={validateNewQuantity}
+              validateSubtraction={validateSubtraction}
+              validateSum={validateSum}
```

Comment:
Remove unnecessary validations and instead, pass one value to the input field and pass a function that will handle setting the value if the value. entered is 0 or greater

Reply:
![image](https://user-images.githubusercontent.com/90296394/135179487-ba05ff7c-5d01-489c-a971-00c8f019b198.png)